src/luarocks/core/cfg.lua

```diff
@@ -347,8 +347,10 @@ local function make_defaults(lua_version, target_cpu, platforms, home)
       defaults.external_deps_patterns = {
          bin = { "?.exe", "?.bat" },
          -- mingw lookup list from http://stackoverflow.com/a/15853231/1793220
-         -- ...should we keep ?.lib at the end? It's not in the above list.
-         lib = { "lib?.dll.a", "?.dll.a", "lib?.a", "cyg?.dll", "lib?.dll", "?.dll", "?.lib" },
+         --~ 2020-01-23: putting ?.lib first allows use of the llvm-mingw32 compiler toolchain
+         --~ https://github.com/mstorsjo/llvm-mingw
+         --~ The LLVM linker doesn't support linking to a dll (GNU ld does).         
+         lib = { "?.lib", "lib?.dll.a", "?.dll.a", "lib?.a", "cyg?.dll", "lib?.dll", "?.dll" },
```

Comment:
Is there a way to check that this change does not affect non-LLVM mingw32 in some way? Not sure if the testsuite (and the CI run for `spec/build_spec.lua` in Appveyor) covers it.

Reply:
> Not sure if the testsuite (and the CI run for spec/build_spec.lua in Appveyor) covers it.

I think there may be a chance it does, in the very latest code of `master`. You might want to rebase this PR over the latest `origin/master` and force-push it so we get a new CI run.

Reply:
On deeper inspection, I don't think the test suite actually checks that the build of a module with an external dependency works correctly (i.e. that it produces a module that the Lua interpreter is really able to load and use). I'm reluctant about merging this before further testing, I can try my hand at extending the tests (probably next week).

Reply:
I got a build working last night so I can move forward and test; no rush on up-streaming it. What platform are you concerned about? Windows, Linux?